posthog/api/test/test_feature_flag.py

```diff
@@ -348,28 +350,194 @@ def test_updating_a_feature_flag_with_same_team_and_key_of_a_deleted_one(self):
+        response = self.client.post(
+            "/api/feature_flag_override",
+            {"feature_flag": feature_flag_instance.id, "user": self.user.id, "override_value": True},
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        response = self.client.get("/api/feature_flag_override")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            len(response.json()["results"]), 1,
+        )
+        self.assertEqual(
+            response.json()["results"][0]["override_value"], True,
+        )
+
+        response = self.client.post(
+            "/api/feature_flag_override",
+            {"feature_flag": feature_flag_instance.id, "user": self.user.id, "override_value": False},
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        response = self.client.get("/api/feature_flag_override")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            len(response.json()["results"]), 1,
+        )
+        self.assertEqual(
+            response.json()["results"][0]["override_value"], False,
+        )
+
+        feature_flag_override_id = response.json()["results"][0]["id"]
+        response = self.client.patch(
+            f"/api/feature_flag_override/{feature_flag_override_id}", {"override_value": True},
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        response = self.client.get("/api/feature_flag_override")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.json()["results"][0]["override_value"], True,
+        )
+
+    def test_create_override_error(self):
+        feature_flag_instance = FeatureFlag.objects.create(team=self.team, created_by=self.user, key="beta-feature")
+        response = self.client.post(
+            "/api/feature_flag_override",
+            {"feature_flag": feature_flag_instance.id, "user": self.user.id, "override_value": {"key": "a dict"}},
+        )
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
-        self.assertEqual(response.json()["type"], "validation_error")
-        self.assertEqual(response.json()["code"], "invalid_input")
-        self.assertEqual(response.json()["detail"], "Please provide a distinct_id to continue.")
+
+    def test_create_override_for_another_team(self):
+        feature_flag_instance = FeatureFlag.objects.create(team=self.team, created_by=self.user, key="beta-feature")
+        team_1_user = self.user
+        _, _, team_2_user = User.objects.bootstrap("Test", "team2@posthog.com", None)
+        self.client.force_login(team_2_user)
+        response = self.client.post(
+            "/api/feature_flag_override",
+            {"feature_flag": feature_flag_instance.id, "user": team_1_user.id, "override_value": True},
```

Comment:
You're testing here both the user & feature flag belonging to another team/org, but it's probably worth testing both cases individually too.